ocean_lib/ocean/ocean_assets.py

```diff
@@ -598,8 +598,6 @@ def create(
         # Fetch the asset on chain
         if wait_for_aqua:
             asset = self._aquarius.wait_for_asset(did)
-        else:
```

Comment:
I would keep the did return here, because otherwise we "loose" it. When not waiting for aqua, if a did is generated, then the user won't know what they created. They can use the did later to wait for aqua or handle in any way, but if we delete this they have no reference to what was created.

Reply:
This change is wrt issue #1087.  The bugs that it is solving are:

- Error 1: it returned the did. create() is supposed to return either the ddo ("asset") or the tuple
- Error 2: it didn't account for whether to return asset vs the tuple

That's why right below it does the "return"

However, I agree, it would be useful to also get the did as part of the tuple. I would consider that a separate issue, because it's changing the interface of create(). I can create that issue and put it as part of this PR. OK?

Reply:
> However, I agree, it would be useful to also get the did as part of the tuple. I would consider that a separate issue, because it's changing the interface of create(). I can create that issue and put it as part of this PR. OK?

I dug into the code more. `create()` already tacks the did onto the ddo. Though it's unclear whether it retains that after the call to aquarius.

Note: I just created #1094 "Improve ocean_assets::create() interface & implementation". It will make create() much better. I plan to do it soon.

Rather than changing did stuff in this PR, let's defer it to that new issue, as part of a bigger cleanup of create().

Then, are you ok with this PR as-is?